File: control-flow/Q112/main.go
package main

import "fmt"

func main() {
	var (
		a int
		b int
		c int
	)
	fmt.Print("Enter a: ")
	fmt.Scan(&a)
	fmt.Print("Enter b: ")
	fmt.Scan(&b)
	fmt.Print("Enter c: ")
	fmt.Scan(&c)
	if a > (b+c) || c > (b+a) || b > (c+a) {
		fmt.Println("NO")
	} else {
		fmt.Println("YES")
	}
}

File: control-flow/Q112/main.rs
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).expect("Failed to read input");
    let mut it = input.split_whitespace();

    println!("Enter a:");
    let a: i32 = match it.next() {
        Some(v) => v.parse().expect("Not a valid number"),
        None => return,
    };

    println!("Enter b:");
    let b: i32 = match it.next() {
        Some(v) => v.parse().expect("Not a valid number"),
        None => return,
    };

    println!("Enter c:");
    let c: i32 = match it.next() {
        Some(v) => v.parse().expect("Not a valid number"),
        None => return,
    };

    if a > b + c || c > b + a || b > c + a {
        println!("NO");
    } else {
        println!("YES");
    }
}